### ISSUE TITLE
Suggestion: Self-contained NuGet utility release

### ISSUE THREAD

**Issue by am11** [NuGet/Home#4079]
With `dotnet` (muxer) utility, it has been made possible to implicitly consume nuget on all platforms that CoreCLR supports.

If such a bootstrapping is possible, please further consider isolating the nuget executable and pack only the essentials parts of runtime/dependencies in a separate release channel for xplat. This will help the non-dotnet consumers effectively depend on nuget utility with minimal footprints.

I assume CoreRT (with its 'compile to native' ability) in build pipeline might be a better tool to distill the binaries as self-contained native executables (at the cost of building native binaries for each platform); I just wanted to highlight the merits of having such a release channel with native or managed self-dependent utility from official authority has potential use cases.

For instance, (perhaps far-stretched example but we can imagine) a Linux distribution using nuget as their package delivery vehicle. :)

**Comment by rrelyea** [comment]
Can you rephrase "If such a bootstrapping is possible, please further consider isolating the nuget executable and pack only the essentials parts of runtime/dependencies in a separate release channel for xplat. This will help the non-dotnet consumers effectively depend on nuget utility with minimal footprints."

Are you asking us to continue to invest on putting more of our support/features into dotnet.exe?

**Comment by am11** [comment]
> Are you asking us to continue to invest on putting more of our support/features into dotnet.exe?

It is actually about self-contained `nuget.exe` which runs cross platform without having to install Mono or dotnet-cli.

Suppose, we wanted to pull C++ headers from VC team's nuget feed and test with LLVM toolchain; in Windows PowerShell we can do:

```sh
# download nuget util
wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile nuget.exe

# install VC build tools in a sub-directory
./nuget install VisualCppTools -source http://vcppdogfooding.azurewebsites.net/nuget/ -Prerelease
```

but for example in Debian I will need to depend on dotnet-cli (which programmatically implements NuGet .NET API) to restore this package.

**Comment by rrelyea** [comment]
Thanks for the request. Will consider feasibility/relative priority in future.

**Comment by jlholtus** [comment]
I'm using Cake Build (C# make) and it bootstraps itself using NuGet. But to run NuGet it depends on mono. I have an interest in getting rid of that dependency and doing exactly what am11 suggested; download the command line util and run it stand-alone.

Are there any blockers that would prevent adding this feature?

**Comment by rrelyea** [comment]
@anangaur - fyi
